### PR TITLE
use conditional return type for is_a() via stub

### PR DESCRIPTION
**Pull request by staabm**
refs https://github.com/phpstan/phpstan/issues/4371#issuecomment-1124893300

### CODE REVIEW

File: stubs/core.stub [hunk 1]
```diff
@@ -1,5 +1,13 @@
 <?php
 
+/**
+ * @param mixed $object_or_class
```
Comment (staabm) [inline stubs/core.stub:4]: couldn't we even use a conditional parameter type here?

```suggestion
 * @param ($allow_string is false ? object : object|class-string) $object_or_class
```

Comment (ondrejmirtes) [reply stubs/core.stub:4]: Try it after this PR is merged :)

Comment (staabm) [reply stubs/core.stub:4]: oh my.. now I get what you was telling me in https://github.com/phpstan/phpstan-src/pull/1306#issuecomment-1125009375 .. will send a 2nd PR ..

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**